src/Module/BaseSettings.php

```diff
@@ -21,116 +21,154 @@
 
 namespace Friendica\Module;
 
+use Friendica\App;
 use Friendica\BaseModule;
 use Friendica\Content\Feature;
+use Friendica\Content\Nav;
+use Friendica\Core\L10n;
 use Friendica\Core\Renderer;
-use Friendica\DI;
+use Friendica\Core\Session\Capability\IHandleUserSessions;
+use Friendica\Core\System;
+use Friendica\Module\Security\Login;
+use Friendica\Network\HTTPException\ForbiddenException;
+use Friendica\Util\Profiler;
+use Psr\Log\LoggerInterface;
 
 class BaseSettings extends BaseModule
 {
-	public static function createAside()
+	/** @var App\Page */
+	protected $page;
+	/** @var IHandleUserSessions */
+	protected $session;
+
+	public function __construct(IHandleUserSessions $session, App\Page $page, L10n $l10n, App\BaseURL $baseUrl, App\Arguments $args, LoggerInterface $logger, Profiler $profiler, Response $response, array $server, array $parameters = [])
+	{
+		parent::__construct($l10n, $baseUrl, $args, $logger, $profiler, $response, $server, $parameters);
+
+		$this->page    = $page;
+		$this->session = $session;
+
+		if ($this->session->getSubManagedUserId()) {
+			throw new ForbiddenException($this->t('Permission denied.'));
+		}
+	}
+
+	protected function content(array $request = []): string
+	{
+		Nav::setSelected('settings');
+
+		if (!$this->session->getLocalUserId()) {
+			System::httpExit(Login::form());
```

Comment:
I like the login redirect with return path much more tbh

Reply:
There's something elegant about Module re-routing which avoids global state variables, but in practice you're probably right.